pixl_dcmd/src/pixl_dcmd/main.py

```diff
@@ -81,344 +81,139 @@ def anonymise_dicom(dataset: Dataset) -> Dataset:
 
     logger.info("Anonymising received instance")
 
-    # Rip out overlays/
-    dataset = remove_overlays(dataset)
-    logger.info("Removed overlays")
-
     # Merge tag schemes
     tag_operations = load_tag_operations(project_config)
-    all_tags = merge_tag_schemes(tag_operations, manufacturer=dataset.Manufacturer)
+    tag_scheme = merge_tag_schemes(tag_operations, manufacturer=dataset.Manufacturer)
 
-    # Apply scheme to instance
-    dataset = apply_tag_scheme(dataset, all_tags)
-    # Apply whitelist
-    dataset = enforce_whitelist(dataset, all_tags)
+    modalities = project_config.project.modalities
 
     logger.info(
-        f"DICOM tag anonymisation applied according to {project_config.tag_operation_files}"
+        f"Applying DICOM tag anonymisation according to {project_config.tag_operation_files}"
     )
-    # Write anonymised instance to disk.
-    return dataset
+    logger.debug(f"Tag scheme: {tag_scheme}")
+
+    if (0x0008, 0x0060) in dataset and dataset.Modality not in modalities:
+        msg = f"Dropping DICOM Modality: {dataset.Modality}"
+        logger.error(msg)
+        raise ValueError(msg)
```

Comment:
Might be worth using a custom Exception type could rename `PixlSkipMessageError` to `PixlDiscardError` or similar?

Reply:
oh good catch